content/source/docs/enterprise/sentinel/import/tfplan.html.md

```diff
@@ -8,258 +8,538 @@ description: |-
+the Terraform _plan_ (as opposed to actual configuration and state).
 
-This is useful for policies that may restrict behaviors depending
-on prior state. For example, a policy that doesn't allow changing
-removing old values once they're set can use this to determine what
-values were set previously.
+[import-tfconfig]: /docs/enterprise/sentinel/import/tfconfig.html
+[import-tfstate]: /docs/enterprise/sentinel/import/tfstate.html
+
+-> Note that these aliases are not represented as maps. While they will appear
+empty when viewed as maps, the specific import namespace keys will still be
+accessible.
+
+-> Note that while current versions of Terraform Enterprise (TFE) source _all_
+data (including configuration and state) from the plan for the Terraform run in
+question, future versions of TFE may source data accessed through the `tfconfig`
+and `tfstate` imports (as opposed to `tfplan.config` and `tfplan.state`) from
+actual config bundles, or state as stored by TFE. When this happens, the
+distinction here will be useful - the data in the aliased namespaces will be the
+config and state data as the _plan_ sees it, versus the actual "physical" data.
+
+### Function: `module()`
+
+```
+module = func(ADDR)
+```
+
+* **Return Type:** A [module namespace](#namespace-module).
+
+The `module()` function in the [root namespace](#namespace-root) returns the
+[module namespace](#namespace-module) for a particular module address.
+
+The address must be a list and is the module address, split on the period (`.`),
+excluding the root module.
+
+Hence, a module with an address of simply `foo` (or `root.foo`) would be
+`["foo"]`, and a module within that (so address `foo.bar`) would be read as
+`["foo", "bar"]`.
+
+As an example, given the following module block:
+
+```hcl
+module "foo" {
+  # ...
+}
+```
+
+If the module contained the following content:
+
+```hcl
+resource "null_resource" "foo" {
+  triggers = {
+    foo = "bar"
+  }
+}
+```
 
-### tfplan.module_paths
 
-All the module paths represented in the diff. This can be used along
-with tfplan.module() to iterate through all modules. An example of
-a function to retrive all resources in all modules is shown below:
+The following policy would evaluate to `true`:
 
 ```python
 import "tfplan"
 
-all_resources = func() {
-    resources = []
-    for tfplan.module_paths as path {
-        resources += values(tfplan.module(path).resources)
-    }
+main = rule { tfplan.module(["foo"]).resources.null_resource.foo[0].applied.triggers.foo is "bar" }
+```
+
+### Value: `module_paths`
+
+* **Value Type:** List of a list of strings.
+
+The `module_paths` value within the [root namespace](#namespace-root) is a list
+of all of the modules within the Terraform _diff_ for the current plan.
+
+Note the distinction here - this means if there are no changes for any resources
+within the module of concern, the module will not show up in `module_paths`.
```

Comment:
It does. I'm re-doing this section to be a little less animated, so that it matches tone with what I will be adding for all 3 of the module instances in each import (ie: an invalid module or module unknown to config/plan/state will return null).